src/main/java/com/example/demo/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,58 @@
+package com.example.demo.controller;
+
+import com.example.demo.dto.shoppingcart.CartItemRequestDto;
+import com.example.demo.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.demo.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/shopping_cart")
+public class ShoppingCartController {
+
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
+    @PostMapping
+    @Operation(summary = "Add book to the shopping cart",
+            description = "Add book to the shopping cart")
+    public ShoppingCartResponseDto addBookToCart(@PathVariable @Valid Long userId,
+             @RequestBody @Valid CartItemRequestDto cartItemRequestDto) {
+        return shoppingCartService.addBookToShopCart(userId, cartItemRequestDto);
```

Comment:
```suggestion
        User user = (User) authentication.getPrincipal();
        return shoppingCartService.addBookToShopCart(user.getId(), cartItemRequestDto);
```

Please, use Spring Security Authentication object to get user
Check all places